onnxruntime/python/tools/transformers/profiler.py

```diff
@@ -58,29 +66,18 @@ def create_inputs(model, batch_size, sequence_length, samples, input_ids_name, s
 
     return all_inputs
 
-
 def run_profile(onnx_model_path,
                 use_gpu,
                 thread_num,
                 batch_size,
                 sequence_length,
-                samples=1,
-                input_ids_name=None,
-                segment_ids_name=None,
-                input_mask_name=None,
-                dummy_inputs=None):
+                all_inputs):
     from benchmark_helper import create_onnxruntime_session
 
     session = create_onnxruntime_session(onnx_model_path, use_gpu, num_threads=thread_num, enable_profiling=True)
```

Comment:
Seems the session will enable all ORT optimization in default so that the model we actually profiling is not the one feeding in. I am wondering is it better to disable all forms of optimizations here? or provide a choice on opt level?

Reply:
Add a new option --basic_optimization.

Reply:
Thanks!